commit be0a1a32207595c0b6dc952f48630f1b077d7aeb
Author: Iain Sandoe <iain@sandoe.co.uk>
Date:   Fri Mar 4 12:34:15 2022 +0000

    Darwin, libgcc: Fix build errors on powerpc-darwin8.
    
    PowerPC Darwin8 is the last version to use an unwind frame fallback routine.
    This had been omitted from the new shared EH library, along with one more
    header dependency that only fires there.
    
    Signed-off-by: Iain Sandoe <iain@sandoe.co.uk>
    
    libgcc/ChangeLog:
    
            * config/rs6000/t-darwin-ehs: Add darwin-fallback.o.
            * config/t-darwin-ehs: Add dependency on unwind.h.
    
    (cherry picked from commit c18ddb05b0391a397f8882fc6a12a1bab7e0df52)

diff --git libgcc/config/rs6000/t-darwin-ehs libgcc/config/rs6000/t-darwin-ehs
index 42f521411af..581344e862a 100644
--- libgcc/config/rs6000/t-darwin-ehs
+++ libgcc/config/rs6000/t-darwin-ehs
@@ -1,3 +1,3 @@
-# We need the save_world code for the EH library.
+# We need the save_world and anu unwind fallback code for the EH library.
 
-LIBEHSOBJS += darwin-world_s.o
+LIBEHSOBJS += darwin-world_s.o darwin-fallback.o
diff --git libgcc/config/t-darwin-ehs libgcc/config/t-darwin-ehs
index 95275023dac..df46f8a6529 100644
--- libgcc/config/t-darwin-ehs
+++ libgcc/config/t-darwin-ehs
@@ -3,5 +3,5 @@
 
 LIBEHSOBJS = unwind-dw2_s.o unwind-dw2-fde-darwin_s.o unwind-c_s.o
 
-unwind-dw2_s.o: gthr-default.h md-unwind-support.h
+unwind-dw2_s.o: gthr-default.h md-unwind-support.h unwind.h
 $(LIBEHSOBJS): libgcc_tm.h
